django/core/validators.py

```diff
@@ -312,6 +312,8 @@ def validate_ipv4_address(value):
 
 
 def validate_ipv6_address(value):
+    if isinstance(value, ipaddress.IPv6Address):
+        return
     if not is_valid_ipv6_address(value):
```

Comment:
I wonder if we should catch `TypeError` here to prevent crashing for any non-string values :thinking:, e.g.
```python
try:
    is_valid_ipv6_address = is_valid_ipv6_address(value)
except TypeError:
    is_valid_ipv6_address = False
if not is_valid_ipv6_address:
    raise ValidationError(
        _("Enter a valid %(protocol)s address."),
        code="invalid",
        params={"protocol": _("IPv6"), "value": value},
    )
```

Reply:
This is a good idea but I would not add this logic here, I think I would move it/encapsulate it inside `utils/ipv6.py` (and update the docstrings there).

Reply:
Updated

Reply:
> I think we can remove the added `if` in the validator

Do you want to move it to the `is_valid_ipv6_address()`?

Reply:
Yes, I'm pushing the change, I had it locally when you applied the latest suggestions (thank you!). I wasn't sure if you'd had availability in the middle of your workday. Let me know what you think!

Reply:
Looks great :+1: Thanks